In [8]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import pathlib
import statsmodels.api as sm

In [9]:
path = pathlib.Path().cwd().parents[0] / 'CSVs' / 'CarInsurance.csv'
df = pd.read_csv(path)
df.head()

Customer_ID Gender  Age  Driving_License  Region_Code  Previously_Insured   
0       381110      M   25                1           11                   1  \
1       381111      M   40                1           28                   0   
2       381112      M   47                1           28                   0   
3       381113      M   24                1           27                   1   
4       381114      M   27                1           28                   1   

  Vehicle_Age Vehicle_Damage  Annual_Premium  Policy_Sales_Channel  Vintage  
0    < 1 Year             No           35786                   152       53  
1    1-2 Year            Yes           33762                     7      111  
2    1-2 Year            Yes           40050                   124      199  
3    < 1 Year            Yes           37356                   152      187  
4    < 1 Year             No           59097                   152      297

In [11]:
X = sm.add_constant(pd.DataFrame(df['Age']))
Y = pd.DataFrame(df['Annual_Premium'])

In [15]:
model = sm.OLS(Y, X).fit()

display(model.summary())

print(f'The linear regression model is: \n Y = {model.params[0]:.3f} + {model.params[1]:.3f} * X')


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:         Annual_Premium   R-squared:                       0.004
Model:                            OLS   Adj. R-squared:                  0.004
Method:                 Least Squares   F-statistic:                     573.5
Date:                Tue, 31 Oct 2023   Prob (F-statistic):          1.87e-126
Time:                        13:28:41   Log-Likelihood:            -1.4170e+06
No. Observations:              127037   AIC:                         2.834e+06
Df Residuals:                  127035   BIC:                         2.834e+06
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const       2.768e+04    128.014    216.204      0.000    2.74e+04    2.79e+04
Age           73.4513      3.067     23.948      0.000      67.440      79.463
==============================================================================
Omnibus:                    44985.572   Durbin-Watson:                   1.997
Prob(Omnibus):                  0.000   Jarque-Bera (JB):          1620310.420
Skew:                           1.041   Prob(JB):                         0.00
Kurtosis:                      20.372   Cond. No.                         113.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

The linear regression model is: 
 Y = 27677.236 + 73.451 * X


The coefficient $73.451$ means that the `Annual_Premium` increases by $\$73.451$ for each additional year of age.

Because the 